demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/SingleMatch.scala

```diff
@@ -1,43 +1,98 @@
 import uk.gov.ons.addressIndex.demoui.model._
-import uk.gov.ons.addressIndex.demoui.views
-
+import uk.gov.ons.addressIndex.demoui.modules.DemouiConfigModule
+import uk.gov.ons.addressIndex.model.{AddressIndexSearchRequest, PostcodeAddressFile}
+import uk.gov.ons.addressIndex.model.AddressScheme._
+import uk.gov.ons.addressIndex.model.server.response.AddressBySearchResponseContainer
 
 @Singleton
-class SingleMatch @Inject()(val configurataion : Configuration, val messagesApi: MessagesApi)(implicit ec : ExecutionContext) extends Controller with I18nSupport {
+class SingleMatch @Inject()(
+   conf : DemouiConfigModule,
+   val messagesApi: MessagesApi,
+   apiClient: AddressIndexClientInstance
+)(implicit ec: ExecutionContext) extends Controller with I18nSupport {
 
   def showSingleMatchPage() : Action[AnyContent] = Action { implicit req =>
     Logger.info("SingleMatch: Rendering Single Match Page")
     // Get language from Config file rather than req.acceptLanguages
-    val defaultLanguage = configurataion.getString("demoui.defaultLanguage")
+    val defaultLanguage = conf.config.defaultLanguage
     Logger.info("SingleMatch: Default Language =  " + defaultLanguage)
     val lang = req.getQueryString("lang").getOrElse(defaultLanguage)
-    messagesApi.setLang(Ok(uk.gov.ons.addressIndex.demoui.views.html.singleMatch(SingleMatch.singleSearchForm)),Lang(lang))
+    messagesApi.setLang(Ok(uk.gov.ons.addressIndex.demoui.views.html.singleMatch(SingleMatch.form,
+      None,
+      None)),
+    Lang(lang))
   }
 
-  def doMatch() = Action { implicit request =>
-    val address = request.body.asFormUrlEncoded.get("address").mkString
-    Ok("matching address not yet implemented - " + address)
+  def doMatch() : Action[AnyContent] = Action.async { implicit request =>
+    val addressText = request.body.asFormUrlEncoded.get("address").mkString
+    val defaultLanguage = conf.config.defaultLanguage
+    val lang = request.getQueryString("lang").getOrElse(defaultLanguage)
+    if (addressText.trim.isEmpty) {
+      Logger.debug("Empty input address")
+      Future.successful(
+        messagesApi.setLang(Ok(uk.gov.ons.addressIndex.demoui.views.html.singleMatch(SingleMatch.form,
+          Some(messagesApi("single.pleasesupply")),
+          None)),
+          Lang(lang))
+      )
+    } else {
+      Future.successful(
+        Redirect(uk.gov.ons.addressIndex.demoui.controllers.routes.SingleMatch.doMatchQS(addressText))
+      )
+    }
   }
 
+  def doMatchQS(input : String) : Action[AnyContent] = Action.async { implicit request =>
+    val addressText = input
+    val defaultLanguage = conf.config.defaultLanguage
+    val lang = request.getQueryString("lang").getOrElse(defaultLanguage)
+     if (addressText.trim.isEmpty) {
+      Logger.debug("Empty input address")
+      Future.successful(
+        messagesApi.setLang(Ok(uk.gov.ons.addressIndex.demoui.views.html.singleMatch(SingleMatch.form,
```

Comment:
Better formatting please
eg:
```
 Obj(
  param1 = inst1,
  param2  = Param2(
    param3 = inst3,
    param4 = inst4
  )  
)
```
notice i use named params here, it's nicer, i recommend named params.